### PR TITLE
add swagger documentation [#159798567]

### PR DESCRIPTION
**Pull request by ajzoell**
Added comments for Swagger and DocFX documentation.

### CODE REVIEW

**Comment by escalonn** [comment]
branch is out of date, merge broker

File: src/ServiceHub.User.Context/Models/Address.cs [hunk 1]
```diff
@@ -14,15 +14,15 @@ public class Address
...
         public string Address1 { get; set; }
-        ///<value> Address line two </value>
+        ///<value> Address line two. </value>
         public string Address2 { get; set; }
         /// <value> The city. </value>
         public string City { get; set; }
         /// <value> The state. </value>
         public string State { get; set; }
-        /// <value> The zip code.. </value>
+        /// <value> The zip code. </value>
```
Comment (escalonn) [inline src/ServiceHub.User.Context/Models/Address.cs]: Inconsistent spacing

Comment (lvonah) [reply src/ServiceHub.User.Context/Models/Address.cs]: Wow!

Comment (ajzoell) [reply src/ServiceHub.User.Context/Models/Address.cs]: I blame @RandallValentin

File: src/ServiceHub.User.Context/Models/Address.cs [hunk 2]
```diff
@@ -14,15 +14,15 @@ public class Address
         /// <value> The unique ID of an address. </value>
         [BsonId]
         public Guid AddressId { get; set; }
-        ///<value> Address line one </value>
+        ///<value> Address line one. </value>
         public string Address1 { get; set; }
-        ///<value> Address line two </value>
+        ///<value> Address line two. </value>
```
Comment (escalonn) [inline src/ServiceHub.User.Context/Models/Address.cs]: for real though let's have consistent spacing or lack thereof after the `///`. whatever is the common style seen in the official docs etc.

**Comment by ajzoell** [comment]
@escalonn Updated the spacing issue

File: src/ServiceHub.User.Context/Models/Name.cs [hunk 3]
```diff
@@ -12,14 +12,14 @@ namespace ServiceHub.User.Context.Models
     /// </remarks>
     public class Name
     {
-        /// <value>The unique Id for the user's Name Object</value>
+        /// <value>The unique Id for the user's Name Object.</value>
```
Comment (escalonn) [inline src/ServiceHub.User.Context/Models/Name.cs]: weird capitalization

File: src/ServiceHub.User.Context/Repositories/IUserRepository.cs [hunk 4]
```diff
@@ -9,10 +9,29 @@ namespace ServiceHub.User.Context.Repositories
...
+        /// <param name="user">The user to be inserted.</param>    
         Task InsertAsync(Context.Models.User user);
+        /// <summary>
+        /// Gets all users.
+        /// </summary>
         Task<List<Context.Models.User>> GetAsync();
+        /// <summary>
+        /// Gets a user by id.
+        /// </summary>
+        /// <param name="id">Gets the user information by id.</param>
```
Comment (escalonn) [inline src/ServiceHub.User.Context/Repositories/IUserRepository.cs]: that's not a description of the id parameter

File: src/ServiceHub.User.Context/Repositories/IUserRepository.cs [hunk 5]
```diff
@@ -9,10 +9,29 @@ namespace ServiceHub.User.Context.Repositories
...
+        /// Updates a user.
+        /// </summary>
+        /// <param name="user">The user to be updated.</param>
         Task UpdateAsync(Context.Models.User user);
+        /// <summary>
+        /// Delete a user by id.
+        /// </summary>
+        /// <param name="id">Id of user we wish to delete.</param>
         Task DeleteAsync(Guid id);
         Task DeleteAllAsync();
```
Comment (escalonn) [inline src/ServiceHub.User.Context/Repositories/IUserRepository.cs]: missing doc for this one

File: src/ServiceHub.User.Service/Controllers/UserController.cs [hunk 6]
```diff
@@ -83,9 +85,10 @@ public async Task<IActionResult> Get(Guid id)
         /// Finds the users by Gender.
         /// </summary>
         /// <param name="gender"></param>
-        /// <returns>all users with the specified gender, or a 400 error
-        /// if the gender isn't valid, or a 500 error if a database error
-        /// occured.</returns>
+        /// <returns>
+        /// All users with the specified gender, or a 400 error
+        /// if the gender isn't valid, or a 500 error if a database error occured.
```
Comment (escalonn) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: spelling

File: src/ServiceHub.User.Testing/Library/Models/AddressTests.cs [hunk 7]
```diff
@@ -11,7 +11,7 @@ public class AddressTests
         readonly Address _address;
 
         #region Test Data
-        /// <value> A string over the max length for Name. </value>
+        /// <value> A string over the max length for Name.</value>
```
Comment (Tallisble) [inline src/ServiceHub.User.Testing/Library/Models/AddressTests.cs]: Space after <value> and before "A" should be removed to be consistent with rest of documentation.

File: src/ServiceHub.User.Testing/Library/Models/AddressTests.cs [hunk 8]
```diff
@@ -11,7 +11,7 @@ public class AddressTests
...
         readonly string OversizedString = new string('A', 256);
         /// <value> A string of max length for Name. </value>
```
Comment (Tallisble) [inline src/ServiceHub.User.Testing/Library/Models/AddressTests.cs]: Space after and before "A" should be removed.

File: src/ServiceHub.User.Testing/Library/Models/AddressTests.cs [hunk 9]
```diff
@@ -11,12 +11,12 @@ public class AddressTests
         readonly Address _address;
 
         #region Test Data
-        /// <value> A string over the max length for Name. </value>
+        /// <value>A string over the max length for Name.</value>
         readonly string OversizedString = new string('A', 256);
-        /// <value> A string of max length for Name. </value>
+        /// <value>A string of max length for Name. </value>
```
Comment (Tallisble) [inline src/ServiceHub.User.Testing/Library/Models/AddressTests.cs]: remove space after "." and before closing value tag

File: src/ServiceHub.User.Testing/Library/Models/AddressTests.cs [hunk 10]
```diff
@@ -11,12 +11,12 @@ public class AddressTests
...
         readonly string MaxLengthString = new string('A', 255);
 
-        /// <value> A generic name that will serve as the baseline for the tests. </value>
+        /// <value>A generic name that will serve as the baseline for the tests. </value>
```
Comment (Tallisble) [inline src/ServiceHub.User.Testing/Library/Models/AddressTests.cs]: remove space after "." and before closing value tag

File: src/ServiceHub.User.Library/Models/Address.cs [hunk 11]
```diff
@@ -12,30 +12,30 @@ namespace ServiceHub.User.Library.Models
     /// </remarks>
     public class Address
     {
-        /// <value> The unique ID of an address. </value>
+        /// <value>The unique Id of an address.</value>
         public Guid AddressId { get; set; }
-        ///<value> Address line one </value>
+        /// <value>Address line one.</value>
```
[the remark this answers is not in the document]
Comment (ajzoell) [reply src/ServiceHub.User.Library/Models/Address.cs]: The space after the first "///" should be there. It's like that is the official documentation.

**Comment by ajzoell** [comment]
I should point out that the QueueFactory and IQueueFactory classes were added earlier today. They don't have documentation due to us waiting on reviews before touching that code.

**Pull request merged**